drivers/hmis/app/models/hmis/ce/referral_enroller.rb

```diff
@@ -51,6 +51,24 @@ def create_enrollment(message)
       referral.update!(target_enrollment: enrollment)
     end
 
+    def delete_wip_enrollment(_message)
+      enrollment = referral.target_enrollment
+      return unless enrollment
+
+      unless enrollment.in_progress?
+        message = "target enrollment #{enrollment.id} has already had intake completed, unable to perform delete_wip_enrollment message"
+        raise message if Rails.env.development?
+
+        Sentry.capture_message(message)
+        return # in non-dev env: return, we are unable to perform the action
+      end
+
+      Hmis::Hud::Enrollment.transaction do
+        referral.update!(target_enrollment: nil)
+        enrollment.destroy!
+      end
```

Comment:
None of the other engine code uses transactions because we expect the caller to run engine steps in a transaction. Checking mutations- `SubmitCeReferralStep`, `StartCeReferralStep`, `CreateCeReferral`,`AssignReferralParticipants`, and `CreateDirectCeReferral` all run engine in a transaction. Nested transactions can get weird, so I think its best to avoid that.

```suggestion

      referral.update!(target_enrollment: nil)
      enrollment.destroy!
```

Reply:
I also noticed while looking, that `CreateDirectCeReferral` does run nested transactions (`Hmis::Ce::Referral.transaction do...` and `opportunity.with_lock do...` both create transactions) so that might be something to look further at to see if we can run that mutation in a single transaction instead of nesting.

Reply:
Sounds good - I'll put that on my list and create a ticket if necessary!